internal/kafka/internal/handlers/validation.go

```diff
@@ -145,21 +208,50 @@ func ValidateKafkaUserFacingUpdateFields(ctx context.Context, authService author
 	}
 }
 
-func ValidateKafkaClaims(ctx context.Context, kafkaRequestPayload *public.KafkaRequestPayload, kafkaRequest *dbapi.KafkaRequest) handlers.Validate {
-	return func() *errors.ServiceError {
-		kafkaRequest = presenters.ConvertKafkaRequest(*kafkaRequestPayload, kafkaRequest)
-		claims, err := auth.GetClaimsFromContext(ctx)
-		if err != nil {
-			return errors.Unauthenticated("user not authenticated")
+func getClaims(ctx context.Context) (jwt.MapClaims, *errors.ServiceError) {
+	claims, err := auth.GetClaimsFromContext(ctx)
+	if err != nil {
+		return nil, errors.Unauthenticated("user not authenticated")
+	}
+	return claims, nil
+}
+
+type ValidateKafkaClaimsOptions func(claims *jwt.MapClaims) *errors.ServiceError
```

Comment:
Why the "options" name?  might `KafkaClaimValidation` be a better name?

Reply:
:+1: , as you prefer then. I thought `KafkaClaimValidation` fit there as it is a validation element that is passed to the ValidateClaims method